packages/fx-core/src/core/collaborator.ts

```diff
@@ -751,3 +848,98 @@ function getPrintEnvMessage(env: string | undefined, message: string) {
       ]
     : [];
 }
+
+function selectAppTypeQuestion(): MultiSelectQuestion {
+  return {
+    name: CollaborationConstants.AppType,
+    title: getLocalizedString("core.selectCollaborationAppTypeQuestion.title"),
+    type: "multiSelect",
+    staticOptions: [
+      {
+        id: CollaborationConstants.AadAppQuestionId,
+        label: getLocalizedString("core.aadAppQuestion.label"),
+        description: getLocalizedString("core.aadAppQuestion.description"),
+      },
+      {
+        id: CollaborationConstants.TeamsAppQuestionId,
+        label: getLocalizedString("core.teamsAppQuestion.label"),
+        description: getLocalizedString("core.teamsAppQuestion.description"),
+      },
+    ],
+  };
+}
+
+async function getCollaborationQuestionNode(inputs: Inputs): Promise<QTreeNode> {
+  const root = new QTreeNode(selectAppTypeQuestion());
+
+  // Teams app manifest select node
+  const teamsAppSelectNode = new QTreeNode(selectTeamsAppManifestQuestion());
+  teamsAppSelectNode.condition = { contains: CollaborationConstants.TeamsAppQuestionId };
+  root.addChild(teamsAppSelectNode);
+
+  // Aad app manifest select node
+  const aadAppSelectNode = new QTreeNode(selectAadAppManifestQuestion());
+  aadAppSelectNode.condition = { contains: CollaborationConstants.AadAppQuestionId };
+  root.addChild(aadAppSelectNode);
+
+  // Env select node
+  const envNode = await selectEnvNode(inputs);
+  if (!envNode) {
+    return root;
+  }
+  envNode.condition = {
+    validFunc: validateEnvQuestion,
+  };
+  teamsAppSelectNode.addChild(envNode);
+  aadAppSelectNode.addChild(envNode);
+
+  return root;
+}
+
+export async function validateEnvQuestion(
+  input: any,
+  inputs?: Inputs
+): Promise<string | undefined> {
+  if (inputs?.env || inputs?.targetEnvName) {
+    return "Env already selected";
```

Comment:
Do we need to put strings in this  function into package.nls.json file?

Reply:
These are not visible to user so I think we don't need to put into nls file.